internal/pkg/coordinator/monitor.go

```diff
@@ -463,15 +463,9 @@ func runUnenroller(ctx context.Context, bulker bulk.Bulk, policyID string, unenr
 		Dur("unenrollTimeout", unenrollTimeout).
 		Msg("giving a grace period to Elastic Agent before enforcing unenrollTimeout monitor")
 
-	grace := time.NewTimer(unenrollTimeout)
-
-	select {
-	case <-grace.C:
-		grace.Stop()
-		// skip to the next loop.
-	case <-ctx.Done():
-		grace.Stop()
-		return
+	if err := waitWithContext(ctx, unenrollTimeout); err != nil {
+		l.Err(err).Dur("unenroll_timeout", unenrollTimeout).
+			Msg("failed to delay the unenroller monitor startup")
```

Comment:
the only error from this function would be "context cancelled", maybe not log it at the Err level